src/dbnode/generated/proto/namespace/namespace.proto

```diff
@@ -15,6 +15,12 @@ message IndexOptions {
     int64 blockSizeNanos = 2;
 }
 
+message SchemaOption {
```

Comment:
We could actually make this a little more future proof with very little extra work if it became

```protobuf
message NamespaceSchemas {
    map<string, SchemaHistory> history = 1;
}

message SchemaHistory {
    repeated Schema schemas = 1;
}

message Schema {
    uint32 version = 1;
    bytes definition = 2;
    string message = 3;
}
```

and then for the initial implementation we can just always put one schema with the name "default" in the map. Its a little bit more work upfront but it will save us a ton of time later not having to message with this again and will also make upgrading existing clusters way easier

Reply:
I guess technically within a given proto definition you can have multiple messages so we have two options of supporting multiple schemas:

1. One proto definition per schema
2. One proto definition per namespace with one message per schema

Any idea what the tradeoffs between the two implementations are? I don't know off the top of my head

Reply:
1 gives us more flexibility. Two schema does not have two share the same file descriptor, if they do, it is totally up to the user.

I think file descriptor can be more useful in nested message situation, there are multiple messages, one of them is the main one, others are just used by the main. in this case, user need to tell us which one is the main message.

Reply:
> We could actually make this a little more future proof with very little extra work if it became
> 
> ```protobuf
> message NamespaceSchemas {
>     map<string, SchemaHistory> history = 1;
> }
> 
> message SchemaHistory {
>     repeated Schema schemas = 1;
> }
> 
> message Schema {
>     uint32 version = 1;
>     bytes definition = 2;
>     string message = 3;
> }
> ```
> 
> and then for the initial implementation we can just always put one schema with the name "default" in the map. Its a little bit more work upfront but it will save us a ton of time later not having to message with this again and will also make upgrading existing clusters way easier

agree, will make it future proof.

Reply:
> I guess technically within a given proto definition you can have multiple messages so we have two options of supporting multiple schemas:
> 
> 1. One proto definition per schema
> 2. One proto definition per namespace with one message per schema
> 
> Any idea what the tradeoffs between the two implementations are? I don't know off the top of my head

I think with proto import, I might have to go with option2, it also map nicely into user repo.